lua/Connector.lua

```diff
@@ -65,7 +66,41 @@ function writebyte(addr, value, domain)
   end
 end
 
-local socket = require("socket.core")
+function get_lua_version()
+    local major, minor = _VERSION:match("Lua (%d+)%.(%d+)")
+    assert(tonumber(major) == 5)
+    if tonumber(minor) >= 4 then
+        return "5-4"
+    end
+    return "5-1"
+end
+
+function get_os()
+    local the_os, ext, arch
+    if package.config:sub(1,1) == "\\" then
+        the_os, ext = "windows", "dll"
+        arch = os.getenv"PROCESSOR_ARCHITECTURE"
+    else
+        -- TODO: macos?
+        the_os, ext = "linux", "so"
+        return the_os, ext, "" -- linux doesn't use/have the arch

```

Comment:
linux code path can/should go through the same if as windows. the .so is in the x64 directory.
```suggestion
        arch = "x86_64" -- TODO: read ELF header from /proc/$PID/exe to get arch
```